lib/MetaCPAN/Script/Favorite.pm

```diff
@@ -0,0 +1,183 @@
+package MetaCPAN::Script::Favorite;
+
+use Moose;
+
+use Log::Contextual qw( :log );
+
+use MetaCPAN::Types qw( Bool Int Str );
+
+with 'MooseX::Getopt', 'MetaCPAN::Role::Script';
+
+=head1 SYNOPSIS
+
+Updates the dist_fav_count field in 'file' by the count of ++ in 'favorite'
+
+=cut
+
+has queue => (
+    is            => 'ro',
+    isa           => Bool,
+    default       => 0,
+    documentation => 'Use the queue for updates',
+);
+
+has age => (
+    is  => 'ro',
+    isa => Int,
+    documentation =>
+        'Update distributions that were voted on in the last X minutes',
+);
+
+has distribution => (
+    is            => 'ro',
+    isa           => Str,
+    documentation => 'Update only a given distribution',
+);
+
+has count => (
+    is  => 'ro',
+    isa => Int,
+    documentation =>
+        'Update this count to a given distribution (will only work with "--distribution"',
+);
+
+sub run {
+    my $self = shift;
+
+    if ( $self->count and !$self->distribution ) {
+        die
+            "Cannot set count in a distribution search mode, this flag only applies to a single distribution. please use together with --distribution DIST";
+    }
+
+    $self->index_favorites;
+    $self->index->refresh;
+}
+
+sub index_favorites {
+    my $self = shift;
+
+    my %recent_dists;
+    my $body;
+
+    if ( $self->distribution ) {
+        $body = {
+            query => {
+                term => { distribution => $self->distribution }
+            }
+        };
+
+    }
+    elsif ( $self->age ) {
+        my $favs = $self->es->scroll_helper(
+            index       => $self->index->name,
+            type        => 'favorite',
+            search_type => 'scan',
+            scroll      => '5m',
+            fields      => [qw< distribution >],
+            size        => 500,
+            body        => {
+                query => {
+                    range => {
+                        date => { gte => sprintf( 'now-%dm', $self->age ) }
+                    }
+                }
+            }
+        );
+
+        while ( my $fav = $favs->next ) {
+            my $dist = $fav->{fields}{distribution}[0];
+            $recent_dists{$dist}++ if $dist;
+        }
+
+        my @keys = keys %recent_dists;
+        if (@keys) {
+            $body = {
+                query => {
+                    terms => { distribution => \@keys }
+                }
+            };
+        }
+    }
+
+    # get total fav counts for distributions
+
+    my %dist_fav_count;
+
+    if ( $self->count ) {
+        $dist_fav_count{ $self->distribution } = $self->count;
+    }
+    else {
+        my $favs = $self->es->scroll_helper(
+            index       => $self->index->name,
+            type        => 'favorite',
+            search_type => 'scan',
+            scroll      => '30s',
+            fields      => [qw< distribution >],
+            size        => 500,
+            ( body => $body ) x !!$body,
```

Comment:
Is this a different way of saying `$body ? ( body => $body) : (),`

Reply:
yes, that's the "enterprise" way :)